src/main/java/org/sunbird/workflow/service/impl/SignUpWFServiceImpl.java

```diff
@@ -0,0 +1,240 @@
+package org.sunbird.workflow.service.impl;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.util.ObjectUtils;
+import org.sunbird.workflow.config.Configuration;
+import org.sunbird.workflow.config.Constants;
+import org.sunbird.workflow.models.*;
+import org.sunbird.workflow.postgres.entity.WfStatusEntity;
+import org.sunbird.workflow.postgres.repo.WfStatusRepo;
+import org.sunbird.workflow.producer.Producer;
+import org.sunbird.workflow.service.SignUpWFService;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class SignUpWFServiceImpl implements SignUpWFService {
+
+    @Autowired
+    private WfStatusRepo wfStatusRepo;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private Configuration configuration;
+
+    @Autowired
+    private RequestServiceImpl requestServiceImpl;
+
+    @Autowired
+    private WorkflowServiceImpl wfService;
+
+    @Autowired
+    private Producer producer;
+
+    Logger log = LogManager.getLogger(WorkflowServiceImpl.class);
+
+    @Override
+    public SBApiResponse upsertWorkFlow(String rootOrg, String org, WfRequest wfRequest) {
+        SBApiResponse response = createDefaultResponse(Constants.SIGNUP_SUPPORT_WORKFLOW_CREATE);
+        String errMsg = validateWfRequest(wfRequest);
+        if (!StringUtils.isEmpty(errMsg)) {
+            response.getParams().setErrmsg(errMsg);
+            response.getParams().setStatus(Constants.FAILED);
+            response.setResponseCode(HttpStatus.BAD_REQUEST);
+            return response;
+        }
+        response = changeStatus(rootOrg, org, wfRequest, response);
+        errMsg = (String) response.getResult().get(Constants.ERROR_MESSAGE);
```

Comment:
What if the row itself is not there for the parameter passed.Include the failure case